### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by zaibon**
The issue describe at #250 has been fixed in https://github.com/dgraph-io/badger/issues/243
So I updated badger lib to the last version again.

### CODE REVIEW

**Comment by iwanbk** [comment]
travis build is failed

**Comment by zaibon** [comment]
Yep can't see why though. Test in client packages reports failure but I don't know see any test that failed

**Comment by iwanbk** [comment]
got same error in 0-disk https://travis-ci.org/zero-os/0-Disk/jobs/282651857

**Comment by iwanbk** [comment]
i added some logging to the test, and it magically print the error

https://travis-ci.org/zero-os/0-Disk/jobs/285895697#L489
```
fatal error: runtime: out of memory
```

**Comment by iwanbk** [comment]
profiling my test didn't produce enough result, as it only uses 99MB, mainly by `badger`.

After some digging, i found that the test won't fail if i skip the main part of the test, calling `DeleteWithMeta`.

It seems that this operation is heavy weight for badger, need to check it out more

**Comment by zaibon** [comment]
@chrisvdg can you also check this branch and reduce the memory allocation where needed in our tests. 
Seems this is the same issue as #274

**Comment by chrisvdg** [comment]
@zaibon , fixed, though definitely check if your tests are still doing what you want them to do!

**Comment by iwanbk** [comment]
> @chrisvdg can you also check this branch and reduce the memory allocation where needed in our tests.
> Seems this is the same issue as #274 

Same comment to #274. the test passed for me when i reduce number of logs.
But i'm not sure if it is the right way to make the test passed, because i'm afraid that it means that the deletion will fail if number of data become big.
(in tlog case, it failed for 100 tlogs, passed for 25 or 50 logs).

On the other hand, travis machine is quite small ( 2GB of RAM). And this test never failed on my 16GB PC.

**Comment by chrisvdg** [comment]
@iwanbk check the RAM consumption when testing with and without race detection, I doubt for production it will fail like it does in our testing as we won't run it with race detection ;)

**Comment by iwanbk** [comment]
> , I doubt for production it will fail like it does in our testing as we won't run it with race detection ;)

Yeah, sure we won't.

I just wondering if the RAM usage will always increased with the number of logs increased?

**Comment by iwanbk** [comment]
please forget my comment, i missed important remarks by @chrisvdg at https://github.com/zero-os/0-stor/issues/274#issuecomment-335831742

File: client/check_test.go [hunk 1]
```diff
@@ -36,15 +36,15 @@ func TestCheck(t *testing.T) {
...
+		DistributionNr:         2,
 		DistributionRedundancy: 1,
 	}
 
 	c, err := getTestClient(policy)
 	require.NoError(t, err, "fail to create client")
 	defer c.Close()
 
-	data := make([]byte, 1204*10)
+	data := make([]byte, 602*10)
```
Comment (iwanbk) [inline client/check_test.go]: not really important comment, i just feel strange with `1204` and `602` here :D

**Pull request merged**